resources/src/commonMain/kotlin/ThemedColorProvider.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+@file:JvmName("CommonThemedColorProvider")
+package com.splendo.kaluga.resources
+
+import kotlin.jvm.JvmName
+import kotlin.reflect.KProperty
+
+/** Multiplatform Color Provider that is aware of a Dark and Light mode */
+abstract class ThemedColorProvider {
+
+    /** light version of this color */
+    abstract val lightColor: Color
+
+    /** dark version of this color */
+    abstract val darkColor: Color
+
+    /** main color, automatically changing based on theme */
+    abstract val color: Color
```

Comment:
I think this API might be improved a bit, if **color** property type would be **Flow<Color>**. That would explicitly indicate its dynamic nature.

Reply:
I see your point, and this feature can be implemented with a Flow.
But, thinking about the use case, I am not sure that is the best way to go.
This provider is meant to unify and simplify the usage of cross-platform color resources, when it comes to Dark/Light mode. Which is quite different in the way that each color on iOS has 2 versions, but not on Android (where a set of colors has 2 versions, not the color itself)
iOS has no need for a Flow, cause the changing of colors on UI is something that happens by using `colorWithDynamicProvider `. I don't see it as something you'd need to collect multiple times.

Also on the android side, I don't see a use case where is important to observe this color change, if not to trigger recomposition.
But even then, is not the change of color that should be observed but rather the Theme change.
So simplicity is preferred in this case.
In some projects using Kaluga, the common color resource file is currently made of getters or lazy initializations.

FYI:
With this idea, on Compose, there is no need to implement Material Colors: is currently intended for colors that we can use directly from a definition in common code. (a Compose Color can be easily created from a Kaluga Color).
Maybe it matters more to indicate the dynamic nature of these assets when referenced on the ViewModels.

Reply:
> But even then, is not the change of color that should be observed but rather the Theme change.

I absolutely agree, but I don't see a shared implementation of `Flow<Theme>` either. Moreover, in Compose UI it won't really work: the re-composition will indeed be triggered, but due to [the skipping if the inputs haven't changed](https://developer.android.com/jetpack/compose/lifecycle#skipping), it might not affect the places, where colors are being read (it really depends on the implementation).